### DISCUSSION TITLE
Rendre invisible un acteur en fonction de l'affichage d'une scene

### DISCUSSION THREAD

**Discussion by Synfactory** [witsa/synapps#737] [Q&A] [answered]
Hello,
bon je me suis cassé les dents pas mal de fois et je pense que ca doit être relativement simple. Je souhaiterai rendre un acteur visible/invisible en fonction d'une scenekey d'un acteur scene.

Merci de vos retours

**Comment by Synfactory** [comment, marked as answer]
Bon j'ai finalement réussi à me dépatouiller en allant chercher la sceneKey dans une additionnelle de mon acteur et en créant un évènements onPropertyChanged :
```javascript
if ((this.additionals.prop1 == 'ExterieurPark') || (this.additionals.prop1 == 'ExterieurMag'))  {
  this.properties.visible = false
} else {
  this.properties.visible = true
};
```

**Reply by JeremieDENAEGEL** [reply]
Bonjour,

Tu pourrais aussi utiliser du OnReadTransform sur la propriété visible de l'acteur avec le script :
return context.value === this.additionals.displaySceneKey;

En liaison de la propriété, tu pointes properties.sceneKey de l'acteur frame
L'additionnelle displaySceneKey correspond à la scène cible
Ainsi si la scène souhaité s'affiche dans la frame, on affiche l'acteur

Cela pourrait t'éviter d'utiliser OnPropertyChanged et ainsi optimiser ta scène.

**Reply by Synfactory** [reply]
Merci Jeremie
